packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx

```diff
@@ -0,0 +1,1547 @@
+      // Only auto-format if input event is not a character being deleted
+      // and not as user is changing value using up / down keys
+      if (
+        inputEvent.inputType !== "deleteContentBackward" &&
+        !this.preventAutoFormatting
+      ) {
+        if (
+          input.value.length === 1 &&
+          ((input === this.dayInputEl && +input.value >= 4) ||
+            (input === this.monthInputEl && +input.value >= 2)) &&
+          +input.value <= 9
+        ) {
+          this.setInputValue(input);
+          this.notifyScreenReader(input, event);
+          this.moveToNextInput(input);
+        }
+
+        if (input.value.length === 2) {
+          if (+input.value === 0) {
+            input.value = "01";
+          }
+          this.setInputValue(input);
+          this.setPreventInput(input, true);
+          this.notifyScreenReader(input, event);
+          this.moveToNextInput(input);
+        } else {
+          this.setPreventInput(input, false);
+        }
+      } else if (this.preventAutoFormatting) {
+        /**
+         * Using arrow keys prevents auto formatting so need to deal with
+         * switching from arrow keys to inputting numbers
+         **/
+        if (input.value.length === 2 && !this.isKeyboardOrEvent(event)) {
+          this.setInputValue(input);
+          this.setPreventInput(input, true);
+          this.moveToNextInput(input);
+        } else {
+          this.setInputValue(input);
+        }
+      }
+
+      if (input.value.length !== 2) {
+        this.setPreventInput(input, false);
+      }
+    } else {
+      if (input.value.length === 4) {
+        this.setInputValue(input);
+        if (
+          Object.prototype.toString.call(event) !== this.EVENT_OBJECT_STRING
+        ) {
+          this.moveToNextInput(input);
+        }
+        this.setPreventInput(input, true);
+      } else {
+        this.setInputValue(input, true);
+        this.setPreventInput(input, false);
+      }
+    }
+
+    // Add / remove class to make input width match size of value i.e. 2 digits' width for day / month, 4 for year
+    this.setFitToValueStyling(input);
+
+    if (input.value.length === 0) {
+      this.setInputValue(input, true);
+      this.setValidationMessage();
+    }
+
+    this.notifyScreenReader(input, event);
+  };
+
+  private handleKeyDown = (event: KeyboardEvent, isInputPrevented: boolean) => {
+    const input = event.target as HTMLInputElement;
+
+    const eventKey = event.key.toLowerCase();
+    // Regex required due to FF allowing all characters as values for number text field.
+    const regex =
+      /-?\d*\.?\d+(e[-+]?\d+)?|[/-]|arrowup|arrowdown|arrowleft|arrowright|shift|tab|backspace|delete/;
+    if (
+      !regex.test(eventKey) &&
+      !(
+        (event.ctrlKey || event.metaKey) &&
+        (eventKey === "v" || eventKey === "c")
+      )
+    ) {
+      event.preventDefault();
+    }
+
+    if (eventKey === "/" || eventKey === "-" || eventKey === ".") {
```

Comment:
potentially could swap this for a switch statement where it's like 
```js
switch(eventKey) {
case "/":
case "-":
case ".":
  event.preventDefault();
  this.setPasteInvalidText();
  this.moveToNextInput(input);
case: "arrowright":
case: "arrowleft":
  event.preventDefault();
  ...
```
but i dunno what you think is more readable!